source/common/http/header_map_impl.cc

```diff
@@ -454,16 +430,15 @@ void HeaderMapImpl::verifyByteSizeInternalForTest() const {
 }
 
 const HeaderEntry* HeaderMapImpl::get(const LowerCaseString& key) const {
-  for (const HeaderEntryImpl& header : headers_) {
-    if (header.key() == key.get().c_str()) {
-      return &header;
-    }
-  }
-
-  return nullptr;
+  return const_cast<HeaderMapImpl*>(this)->getExisting(key);
```

Comment:
That's great that the API for `get` is more consistent now and can also return O(1) headers.
I'm a bit worried about a previous misuse of this method (i.e., assuming that an O(1) header won't be returned). That said, if someone used it improperly, they should update their code.

Reply:
The previous version of get() did return O(1) headers, it just always did a linear scan. It now attempts a trie lookup first to avoid the linear scan. I will add more comments.

Reply:
You're correct, I forgot the updates to `headers_` in `maybeCreateInline` and `removeInline`.